app/controllers/supervisor/base_controller.rb

```diff
@@ -0,0 +1,18 @@
+class Supervisor::BaseController < ApplicationController
+
+  before_action :must_be_supervisor
+  helper_method :supervised_teams
+
+  protected
+
+  def must_be_supervisor
+    unless signed_in? && current_user.roles.includes?('supervisor')
```

Comment:
This would also allow this method to be re-used in the view(s) that currently also have this supervisor check.